### PR TITLE
S3 generate grid

### PR DESCRIPTION
**Pull request by mclack**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
+
+//removes grid container from document flow
+document.getElementById('gridContainer').style.display="none"
+document.getElementById('gameTitle').style.display="none"
```
Comment (mporam) [inline index.js]: why not do this in the css?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+
+//function to generate all game tiles
+document.querySelector('.form').addEventListener('submit', (e) => {
+    //stops form from submitting
+    e.preventDefault()
+    //Defining total grid size, number of lynx and misses from user input
+    let numLynx = parseInt(document.getElementById('numLynx').value)
+    let gridSize = parseInt(document.getElementById('gridSize').value)
+    let miss = gridSize - numLynx
```
Comment (mporam) [inline index.js]: these could be `const`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    //reset
+    document.getElementById('gridContainer').innerHTML = ''
+    document.getElementById('errorMessage').textContent = ''
+
+
+        let generateSquares = () => {
```
Comment (mporam) [inline index.js]: why the extra indentation?

Comment (mporam) [reply index.js]: could also be a `const`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            //create empty array
+            let squares = []
+            //For every hit we are putting a div into the array with data-hit 1
+            for (i = 0; i < numLynx; i++) {
+                squares.push('<div class="grid-item" data-hit="1"></div>')
+            }
+            //For every hit we are putting a div into the array with data-hit 0
```
Comment (mporam) [inline index.js]: incorrect comment. should be, for every **miss**

File: index.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            for (i = 0; i < miss; i++) {
+                squares.push('<div class="grid-item" data-hit="0" ></div>')
+            }
+            //shuffle the array
+            let shuffle = (array) => array.map(a => ({
```
Comment (mporam) [inline index.js]: should be a const

File: index.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                sort: Math.random(),
+                value: a
+            })).sort((a, b) => a.sort - b.sort).map(a => a.value)
+            let shuffledArr = shuffle(squares)
```
Comment (mporam) [inline index.js]: should be a const

File: index.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+            //adding each array element to html of container
+            shuffledArr.forEach((square) => {
+                document.getElementById('gridContainer').innerHTML += square
+            })
+        }
+
+    //validates form input
+    if(!validateInput(gridSize, numLynx)){
+        document.getElementById('errorMessage').textContent = errorMessage
+    }else {
```
Comment (mporam) [inline index.js]: missing space after `}`

File: index.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+            })).sort((a, b) => a.sort - b.sort).map(a => a.value)
+            let shuffledArr = shuffle(squares)
+            //adding each array element to html of container
+            shuffledArr.forEach((square) => {
+                document.getElementById('gridContainer').innerHTML += square
+            })
+        }
+
+    //validates form input
+    if(!validateInput(gridSize, numLynx)){
```
Comment (mporam) [inline index.js]: missing space after `if` and another one between `){`

File: index.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        document.getElementById('errorMessage').textContent = errorMessage
+    }else {
+        //execute function
+        generateSquares()
+        turnCard()
```
Comment (mporam) [inline index.js]: why is generate Squares a anonymous function inside the event listener but turnCard is a named function outside? seems inconsistent. same with the shuffle function.

File: index.js [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+//num2 is the number of lynx.
+//Function checks if number is integer,
+// if positive
+// and if hitboxes (num2) does not exceed half of the grid size (num1).
+// global error message is populated upon error, dictated by the conditional, ready to be displayed
+//return true if no errors
+
+let errorMessage = ""
+
+function validateInput(num1, num2) {
```
Comment (mporam) [inline index.js]: if you name the parameters better you wont need so much description. calling them `num1` and `num2` is not helpful

File: index.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+// Both .hit and .miss are styled in CSS, ready for application.
+// DOCBLOCK ABOVE
+
+gridItems.forEach( (item) => {
+    item.addEventListener('click', () => {
+        if(item.dataset.hit === '1'){
+            item.style.backgroundImage = "url('project-assets/babylynx2.jpg')"
+            item.textContent = 'Hooray, you\'ve found a lynx!'
+        }
+        if(item.dataset.hit === '0'){
```
Comment (mporam) [inline index.js]: why not use an else here? there isn't any other value that hit could be so no need to check it again

File: index.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+// DOCBLOCK BELOW
+// Does a forEach over gridItems and checks the data-hit value for each of them
+// When clicked, if data-hit is 1: Applies class='hit' and changes textContent to 'HIT'
+// When clicked, if data-hit is 0: Applies class='miss' and changes textContent to 'MISS'
+// Both .hit and .miss are styled in CSS, ready for application.
+// DOCBLOCK ABOVE
+
+gridItems.forEach( (item) => {
+    item.addEventListener('click', () => {
+        if(item.dataset.hit === '1'){
```
Comment (mporam) [inline index.js]: please fix spacing:
```
if (condition) {
```

File: index.js [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+    }
+    if ((num1 / 2) < num2) {
+        errorMessage = 'Number of lynx can not exceed half of the number you picked for the Forest size'
+        return false
+    }
+
+    return true
+}
+
+function turnCard() {
```
Comment (mporam) [inline index.js]: this entire function is not indented correctly

File: index.js [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+// Grabs all grid-item divs as gridItems to be used in forEach
+let gridItems = document.querySelectorAll('.grid-item')
+// DOCBLOCK BELOW
+// Does a forEach over gridItems and checks the data-hit value for each of them
+// When clicked, if data-hit is 1: Applies class='hit' and changes textContent to 'HIT'
+// When clicked, if data-hit is 0: Applies class='miss' and changes textContent to 'MISS'
+// Both .hit and .miss are styled in CSS, ready for application.
+// DOCBLOCK ABOVE
```
Comment (mporam) [inline index.js]: is this necessary? code seems pretty self explanatory to me

File: styles.css [hunk 15]
```diff
@@ -8,9 +8,10 @@ body {
 }
 
 header {
-    height: 100vh;
+    /*height: 100vh;*/
```
Comment (mporam) [inline styles.css]: if you dont need this, please remove it

File: styles.css [hunk 16]
```diff
@@ -35,4 +36,78 @@ h2 {
...
+    flex-flow: row wrap;
+    justify-content: space-around;
+}
+
+.grid-item {
+    background-image: url("project-assets/ren-ran-bBiuSdck8tU-unsplash.jpg");
+    background-size: cover;
+    height: 200px;
+    width: 200px;
+    margin: 10px 20px 10px 20px;
```
Comment (mporam) [inline styles.css]: this can be shortened to:
```
margin: 10px 20px;
```

**Review by mporam — DISMISSED** [review]

File: index.js (hunk 1 above)
Comment (HannaSBarratt) [reply index.js]: Done

**Review by HannaSBarratt — COMMENTED** [review]
Made changes in line with Mike's comments

**Review by charliecog — APPROVED** [review]

**Pull request merged**